Sprint 4/step 1

abobkov2009 opened the pull request.

Commit:
Привел наименование всех типов к единому стандарту
Обявление и экспорт функций, так же привел к общему стандарту во всём приложении

Commit:
Объединил некоторые типы аргументов auth (Сделал через Pick<T,List of properties>
Переделал объявление функциональных компонентов по рекоммендациям с вебинара

kyzinatra commented on src/components/alert-message/AlertMessage.tsx:
```diff
@@ -1,13 +1,14 @@
+export default function AlertMessage({ header, message }: TAlertMessageProps): React.JSX.Element {
     return (
         <div className='pt-15'>
             {header && (<h1 className="text text_type_main-large mb-3">{header}</h1>)}
```
Можно лучше: на странице может быть только один `h1` заголовок

kyzinatra commented on src/components/modal/modal.tsx:
```diff
@@ -8,14 +8,14 @@ import ModalOverlay from '../modal-overlay/modal-overlay';
+type TModalProps = {
     onModalClose?: () => void;
     children: React.ReactElement;
 };
```
Можно лучше: `children` можно типизировать как `PropsWithChildren<>`

kyzinatra commented on src/components/burger-constructor/stuffing-element/stuffing-element.tsx:
```diff
@@ -6,19 +6,19 @@ import { ConstructorElement, DragIcon } from '@ya.praktikum/react-developer-burg
-export default function StuffingElement({ ingredient, index }: StuffingElementProps) {
+export default function StuffingElement({ ingredient, index }: TStuffingElementProps): React.JSX.Element {
     const dispatch = useAppDispatch();
     const ref = useRef<HTMLLIElement>(null)
```
Отлично! Хорошая типизация хуков

kyzinatra commented on src/services/auth/api.ts:
```diff
@@ -45,7 +32,7 @@ export const authApi = createApi({
         }),
-        logoutUser: builder.mutation<authResponseType, refreshAndLogoutTokenArgsType>({
+        logoutUser: builder.mutation<TAuthResponse, Pick<TAuthArgs, "token">>({
             query(body) {
```
Отлично! Понравилось, как вы типизировали `redux`

kyzinatra commented on src/services/auth/slice.ts:
```diff
@@ -16,7 +16,7 @@ export const authSlice = createSlice({
-        userLogedIn: (state, action: PayloadAction<userInfoType>) => {
+        userLogedIn: (state, action: PayloadAction<TUserInfo>) => {
             state.user = action.payload;
         },
```
Можно лучше: опечатка `userLogedIn` -> `userLoggedIn`

kyzinatra commented:
Выглядит очень круто! Видно, что проделана большая работа. Типизация redux и всего проекта сделано отлично. Но все же есть моменты на которые стоит обратить внимание.
Подробнее: 


**Можно лучше:**

-  на странице может быть только один `h1` заголовок
-  `children` можно типизировать как `PropsWithChildren<>`
-  опечатка `userLogedIn` -> `userLoggedIn`


**Что понравилось:**

-  Хорошая типизация хуков
-  Понравилось, как вы типизировали `redux`

Commit:
правки "Можно сделать лучше"
- на странице может быть только один h1 заголовок
- children можно типизировать как PropsWithChildren<>
- опечатка userLogedIn -> userLoggedIn
